FWCore/Framework/interface/EventSetupRecord.h

```diff
@@ -180,13 +171,30 @@ namespace edm {
           */
       unsigned long long cacheIdentifier() const { return impl_->cacheIdentifier(); }
 
+      /**When you are processing multiple validity intervals concurrently,
+          each must have its own cache for data. These caches are numbered
+          from 0 to one less than the maximum number of concurrent validity
+          intervals allowed for that record. This number is returned by
+          the following function. For one record type, all the different validity
+          intervals being processed at the same time will have a different
+          iovIndex. But the iovIndex's of record types that are different
+          are independent of each other.
+          */
+      unsigned int iovIndex() const { return impl_->iovIndex(); }
+
       ///clears the oToFill vector and then fills it with the keys for all registered data keys
       void fillRegisteredDataKeys(std::vector<DataKey>& oToFill) const { impl_->fillRegisteredDataKeys(oToFill); }
 
+      static constexpr bool allowConcurrentIOVs_ = true;
```

Comment:
I added a comment to explain this. A derived class can redefine this. This will override the general configuration parameter that sets the number of concurrent IOVs for all records. If redefined false, there will only be 1 IOV allowed at a time for that record.